### PR TITLE
fix($drag): offset dragged element relative to first positioned ancestor

### LINKED ISSUE #63: Draggables within positioned containers
Using position:absolute for draggables doesn't seem to work for elements with positioned ancestors. The element is offset from the cursor by the ancestor's offset.
This happens because the draggables final position is the calculated offset (i.e. event position) plus ancestor offset.

Is this a known issue?
Any particular reason fixed positioning was not used during drag?

Thanks.

### PR DESCRIPTION
**Pull request by adam77**
fixes #63 
Notes:
* $dndPosition is a stripped down version of $position from angular-ui/bootstrap.
* Removed draggable.positionAbs as it didn't appear to be in use.
* Fixed a bug in draggable when using jqlite that showed up in the new test (relevant jqlite bug: https://github.com/angular/angular.js/pull/11614).
* Manually tested OK in IE9.

At some point the DOM utils stuff could be merged with $dndPosition perhaps.

### CODE REVIEW

**Comment by adam77** [comment]
heh, can't argue with those coverage stats.

**Comment by caitp** [comment]
definitely, that's awesome. Let me take a look at this

File: src/position.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                top: elBCR.top - offsetParentBCR.top,
+                left: elBCR.left - offsetParentBCR.left
+            };
+        },
+
+        /**
+         * Provides read-only equivalent of jQuery's offset function:
+         * http://api.jquery.com/offset/
+         */
+        offset: function (element) {
```
Comment (caitp) [inline src/position.js]: I was previously putting helpers like this in src/utils.js --- if moving them here and refactoring, please delete the old dead code and make sure this new code is being used consistently

File: src/public.js [hunk 2]
```diff
@@ -10,5 +10,7 @@ function publishExternalAPI() {
   }]).directive({
     draggable: draggableDirective,
     droppable: droppableDirective
+  }).factory({
+    $dndPosition: $positionFactory
```
Comment (caitp) [inline src/public.js]: I might suggest renaming this to $dndDOM or something, because it will probably do more than just deal with position

**Comment by caitp** [comment]
something got messed up with the coveralls account --- could you rebase and re-push? I just want to see the accurate coverage info before landing

**Comment by caitp** [comment]
the coverage looks okay (non-regression at least), although for whatever reason coveralls is not reporting the new file on their UI (they seem to be suffering from some new bugs lately!)

**Comment by adam77** [comment]
i noticed the new travis file has node v0.1?

**Comment by caitp** [comment]
it's the same as 0.10, it's just formatted differently

**Comment by caitp** [comment]
The demo still works nicely, so as far as I'm concerned it's safe to merge --- it would be good to add to the demo app with a little showcase of your fix. LGTM, landing

**Pull request closed without merging**